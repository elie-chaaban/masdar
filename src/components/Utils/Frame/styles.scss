.nxn-logo-div {
  position: absolute;
  top: 0;
  left: 40px;
  width: 200px;
  height: 60px;
  background-color: #627d98;
  z-index: 999999;
  display: flex;
  justify-content: center;
  align-items: center;
  .nxn-logo-image {
    width: 80px;
  }
}
