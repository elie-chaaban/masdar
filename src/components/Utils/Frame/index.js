import React from 'react';
import PropTypes from 'prop-types';
import logo from '../../../assets/images/logo/logo.svg';
import './styles.scss';

const Frame = ({source, showLogoDiv}) => {
  return (
    <>
      {showLogoDiv && (
        <div className="nxn-logo-div">
          <img className="nxn-logo-image" src={logo} alt="logo"></img>
        </div>
      )}
      <iframe src={source} frameBorder={'0'} style={{width: '100%', height: '100vh'}} title={'iframe'} />
    </>
  );
};

Frame.propTypes = {
  source: PropTypes.string.isRequired
};
export default Frame;
