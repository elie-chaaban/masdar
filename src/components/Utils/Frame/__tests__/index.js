import React from 'react';
import {shallow} from 'enzyme';
import Frame from '..';

test('Frame', () => {
  const dom = shallow(<Frame source="url" />);
  expect(dom.find('iframe').prop('src')).toBe('url');
  expect(dom).toMatchSnapshot();
});
