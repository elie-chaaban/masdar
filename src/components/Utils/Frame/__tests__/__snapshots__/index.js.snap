// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frame 1`] = `
<Fragment>
  <iframe
    frameBorder="0"
    src="url"
    style={
      Object {
        "height": "100vh",
        "width": "100%",
      }
    }
    title="iframe"
  />
</Fragment>
`;
